src/main/java/sparksoniq/spark/iterator/flowr/ReturnClauseSparkIterator.java

```diff
@@ -49,8 +60,53 @@ public ReturnClauseSparkIterator(RuntimeTupleIterator child, RuntimeIterator exp
         return itemRDD;
     }
 
-    private JavaRDD<Item> itemRDD;
-    
-    private RuntimeTupleIterator _child;
+    @Override
+    protected boolean hasNextLocal() {
+        return _hasNext;
+    }
+
+    @Override
+    protected Item nextLocal() {
+        if(_hasNext == true){
+            Item result = _nextLocalResult;  // save the result to be returned
+            setNextLocalResult();            // calculate and store the next result
+            return result;
+        }
+        throw new IteratorFlowException("Invalid next() call in Object Lookup", getMetadata());
+    }
+
+    @Override
+    protected void openLocal(DynamicContext context) {
+        this._currentDynamicContext = context;
+
+        _iterator.open(context);
```

Comment:
I think you shouldn't open _iterator here, but in setNextLocalResult. This is because there will be one dynamic context passed to it for every tuple in the child clause :-) (see explanations below)